source/slang/slang-ir-explicit-global-context.cpp

```diff
@@ -227,6 +336,27 @@ struct IntroduceExplicitGlobalContextPass
         {
             replaceUsesOfGlobalVar(globalVar);
         }
+
+        // SPIRV requires a correct IR func-type to emit properly
+        if (m_options.requiresFuncTypeCorrection())
+        {
+            for (auto pairOfFuncs : m_mapFuncToContextPtr)
+            {
+                if (pairOfFuncs.second->getOp() == kIROp_Var)
+                    continue;
+                auto func = pairOfFuncs.first;
+                builder.setInsertBefore(func->getDataType());
+                auto paramCount = func->getParamCount();
+                List<IRType*> paramTypes;
+                paramTypes.reserve(paramCount);
+                for (UInt i = 0; i < paramCount; i++)
+                    paramTypes.add(func->getParamType(i));
+                paramTypes.add(pairOfFuncs.second->getDataType());
+                auto newFuncType = builder.getFuncType(paramTypes, func->getResultType());
+
+                func->setFullType(newFuncType);
+            }
```

Comment:
I am not sure why this is needed.
But I am worried that you are modifying the key for the map variable, `m_mapFuncToContextPtr`.
It may invalidate the map variable.

Reply:
Should be able to call `fixUpFuncType` for this?
The logic around map is fine here because this logic is modifying the value pointed to by `func`, where the map's key is a pointer to func.

Reply:
> I am not sure why this is needed.

I will update the comment I added to be more descriptive.

> But I am worried that you are modifying the key for the map variable, m_mapFuncToContextPtr.
> It may invalidate the map variable.

The key is the pointer, not the data inside the pointer?